File: DSA LEVEL 1/Recursion with arraylists/get kpc.py
import sys
sys.setrecursionlimit(10**6)
def prn(s,d):
    if s=="":
        return [""]
    f=s[0] 
    res=s[1:]
    ans=prn(res,d)
    ch=d[f]
    m=[]
    for i in ch:
        for j in ans:
            m.append(i+j)
    return m







if __name__ == "__main__":
   s=input()
   d={"0":".;","1":"abc","2":"def","3":"ghi","4":"jkl","5":"mno","6":"pqrs","7":"tu","8":"vwx","9":"yz"}
   a=prn(s,d)
#    a.append(s)
   print(a)

File: DSA LEVEL 1/Recursion with arraylists/get kpc.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
//		int n = Integer.parseInt(br.readLine());
//		int[] arr = new int[n];
//		for (int i = 0; i < n; i++) {
//			arr[i] = Integer.parseInt(br.readLine());
//		}
//
//		int x = Integer.parseInt(br.readLine());
//		int ans=firstIndex(arr, 0, x);
//		
//		System.out.println(ans);
//		displayArr(arr,0);

		String str = br.readLine();
		ArrayList<String> ans = getKPC(str);
		System.out.println(ans);

	}

	static String[] arr = { ".;", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwx", "yz" };

	public static ArrayList<String> getKPC(String str) {
		
		if (str.length()==0) {
			ArrayList<String> br = new ArrayList<>();
			br.add("");
			return br;
		}

		char ch = str.charAt(0);
		String ros = str.substring(1);

		ArrayList<String> rres = getKPC(ros);

		ArrayList<String> mres = new ArrayList<String>();

		String charCode = arr[ ch-'0'];

		for (int i = 0; i < charCode.length(); i++) {
			char chcode = charCode.charAt(i);
			for (String res : rres) {

				mres.add(chcode + res);
			}

		}

		return mres;
	}


}
